File: MTWDM iOS Xamarin/App10/App10/ViewController.cs
using System;
using Foundation;
using UIKit;

namespace App10
{
    public partial class ViewController : UITableViewController
    {

        //DataSource
        string[] datos = { "Leche", "Manzanas", "Jamón", "Huevos" };
        string[] datosNube = { "Panqueques", "Jugo de naranja", "Leche", "Manzanas", "Jamón", "Waffles", "Huevos" };


        //Control para actualizar datos de nuestra TableView
        UIRefreshControl pullToRefreshControl = new UIRefreshControl();


        protected ViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            // Perform any additional setup after loading the view, typically from a nib.

            this.NavigationItem.LeftBarButtonItem = this.EditButtonItem;
            this.NavigationItem.RightBarButtonItem = new UIBarButtonItem(UIBarButtonSystemItem.Add,
                                                                         this,
                                                                         null);

            this.RefreshControl = pullToRefreshControl;
            pullToRefreshControl.AddTarget((sender, args) => refreshTable(), UIControlEvent.ValueChanged);
      
        }

        public void refreshTable()
        {
            datos = datosNube;
            TableView.ReloadData();
            pullToRefreshControl.EndRefreshing();
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }


        public override nint NumberOfSections(UITableView tableView)
        {
            return 1;
        }

        public override nint RowsInSection(UITableView tableView, nint section)
        {
            return datos.Length;
        }

        public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
        {

            var cell = tableView.DequeueReusableCell("comestiblesCell", indexPath);

            if (cell == null)
            {
                cell = new UITableViewCell(UITableViewCellStyle.Default, "comestiblesCell");
            }

            cell.TextLabel.Text = datos[indexPath.Row];

            return cell;

        }
    }
}

File: MTWDM iOS Swift /app10/app10/ViewController.swift
//
//  ViewController.swift
//  app10
//
//  Created by Felipe Hernandez on 10/19/18.
//  Copyright © 2018 Felipe Hernandez. All rights reserved.
//

import UIKit

class ViewController: UITableViewController {

    // MARK: - View Did Load
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "Tabla"
        
        self.navigationItem.leftBarButtonItem = self.editButtonItem
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem:  .add,target: self, action: nil)
        self.refreshControl = pullToRefreshControl
        pullToRefreshControl.addTarget(self, action: #selector(refreshTable), for: .valueChanged)
    }
    
    // MARK: - propiedades
    var datos = ["Leche", "Manzanas", "Jamón", "Huevos"]
    var datosNube = ["Panqueques", "Jugo de naranja", "Leche", "Manzanas", "Jamón", "Waffles", "Huevos"]
    
    let pullToRefreshControl = UIRefreshControl()
    
    // MARK: - funciones
    @objc func refreshTable() {
        datos = datosNube
        self.tableView.reloadData()
        pullToRefreshControl.endRefreshing()
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return datos.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "comestiblesCell", for: indexPath)
        
        cell.textLabel?.text = datos[indexPath.row]
        
        return cell
    }

}

